packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,224 @@
+ */
+export let showCopyButton = true;
+
+/**
+ * Some prism languages have dependencies. For example, C++ requires C. If the
+ * passed `language` has dependencies, they must be included here to be loaded.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let dependencies: string[] = [];
+
+/** Additional CSS classes to pass to the aside. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let element: HTMLElement | undefined;
+
+$: copyButtonLabel = 'Copy';
+$: copyButtonIcon = 'content-copy' as IconName;
+
+const dispatch = createEventDispatcher<{
+  copy: { succeeded: boolean; message: string };
+}>();
+
+const copyToClipboard = async () => {
+  try {
+    await navigator.clipboard.writeText(code);
+
+    copyButtonLabel = 'Copied';
+    copyButtonIcon = 'check';
+
+    dispatch('copy', {
+      succeeded: true,
+      message: 'Successfully copied snippet to the clipboard',
+    });
+  } catch {
+    copyButtonLabel = 'Failed';
+    copyButtonIcon = 'alert';
+
+    dispatch('copy', {
+      succeeded: false,
+      message: 'Failed to copy snippet to the clipboard',
+    });
+  }
+
+  window.setTimeout(() => {
+    copyButtonLabel = 'Copy';
+    copyButtonIcon = 'content-copy';
+  }, 2000);
+};
+
+const highlight = () => {
+  if (element) {
+    Prism.highlightElement(element);
+  }
+};
+
+const formatCode = (input: string): string => {
+  const htmlEntities: Record<string, string> = {
+    '<': '&lt;',
+    '>': '&gt',
+    '/': '&#47;',
+  };
+
+  let formattedCode = input;
+  for (const [key, value] of Object.entries(htmlEntities)) {
+    if (formattedCode) {
+      formattedCode = formattedCode.replaceAll(key, value);
+    }
+  }
+
+  return formattedCode;
+};
+
+$: {
+  if (element && code) {
+    element.innerHTML = formatCode(code);
+    highlight();
+  }
+}
+
+onMount(async () => {
+  try {
+    /**
+     * Load the themes, for vite to properly discover them we cannot use string templates,
+     * so we use a switch statement for our supported themes.
+     */
+    switch (theme) {
+      case 'vs': {
+        await import('prism-themes/themes/prism-vs.min.css');
+        break;
+      }
+      case 'vsc-dark-plus': {
+        await import('prism-themes/themes/prism-vsc-dark-plus.min.css');
+        break;
+      }
+      default: {
+        // Unsupported theme
+        break;
+      }
+    }
+
+    /**
+     * After the HTML is loaded in the DOM, we can use the autoloader to manage
+     * scanning for languages and including the components properly
+     */
+    await import(
+      // @ts-expect-error no type declaration for this JS file
+      'prismjs/plugins/autoloader/prism-autoloader'
+    );
+
+    // Make sure the autoloader knows where to find our languages
+    (Prism.plugins.autoloader as { languages_path: string }).languages_path =
+      // TODO: do we want a better way to get the version?
+      'https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/';
```

Comment:
So I am using the CDN here because otherwise we would have to include all of the potential languages we want to support along with our bundle, which does not seem ideal. One downside here is we are including the version number for prismjs in the CDN link, which means bumping our dev dependency on it could cause issues. Interested in suggestions for a better way to manage that.

Reply:
In the old version I'd import the version from package.json:

```ts
import pkg from '../../package.json';

const version = pkg.devDependencies.prismjs.replace('^', '');
```

That way we still have our package.json as a versioning source of truth

Reply:
So I tried it out and maybe I am missing something, but it works fine for the playground dev environment but fails in the storybook with this error:

```shell
Failed to resolve import "../../package.json" from "../core/dist/code-snippet.svelte". Does the file exist?
3:08:54 PM [vite] Internal server error: Failed to resolve import "../../package.json" from "../core/dist/code-snippet.svelte". Does the file exist?
  Plugin: vite:import-analysis
  File: /Users/devintcurrie/Projects/viam-prime/packages/core/dist/code-snippet.svelte:18:16
  29 |  const { Object: Object_1, console: console_1 } = globals;
  30 |  import "svelte/internal/disclose-version";
  31 |  import pkg from "../../package.json";
  ```

Reply:
@micheal-parks So I actually stole the solution we use for getting the monaco version for the code editor:

```ts
import PrismPackage from 'prismjs/package.json';

...

const PRISM_VERSION = PrismPackage.version;
```